indigo_app/views/works.py

```diff
@@ -1005,3 +1008,217 @@ def get_context_data(self, **kwargs):
+        context["work"] = work = Work(country=self.country, locality=self.locality, frbr_uri="/akn/za/act/2009/1")
+        form = self.update_work(work)
+
+        # use a fresh form based on the partially-updated work to re-render the template
+        context["form"] = self.refreshed_form(form, work)
+
+        return context
+
+    def update_work(self, work):
+        # update the object with the submitted form data, without saving the changes
+        form = self.Form(self.request.GET, instance=work)
+        form.is_valid()
+        return form
+
+    def refreshed_form(self, form, work):
+        return self.Form(instance=work)
+
+
+class WorkFormRepealView(PartialWorkFormView):
+    """Just the repeal part of the work form to re-render the form when the user changes the repeal status
+    through HTMX.
+    """
+    template_name = 'indigo_api/_work_repeal_form.html'
+
+    class Form(forms.ModelForm):
+        prefix = 'work'
+
+        class Meta:
+            model = Work
+            fields = ('repealed_by', 'repealed_date')
+            exclude = ('frbr_uri',)
+
+    def update_work(self, work):
+        form = super().update_work(work)
+        if work.repealed_by:
+            work.repealed_date = (work.repealed_by.commencement_date or
+                                  work.repealed_by.publication_date)
+        return form
+
+
+class WorkFormParentView(PartialWorkFormView):
+    """Just the parent part of the work form to re-render the form when the user changes the parent work through HTMX.
+    """
+    template_name = 'indigo_api/_work_parent_form.html'
+
+    class Form(forms.ModelForm):
+        prefix = 'work'
+
+        class Meta:
+            model = Work
+            fields = ('parent_work',)
+
+
+class WorkFormCommencementView(PartialWorkFormView):
+    """Just the commencing work part of the work form to re-render the form when the user changes the commencing
+     work through HTMX.
+    """
+    template_name = 'indigo_api/_work_commencement_form.html'
+
+    class Form(forms.ModelForm):
+        commencing_work = forms.ModelChoiceField(queryset=Work.objects, required=False)
+        prefix = 'work'
+
+        class Meta:
+            model = Work
+            fields = ('commencing_work',)
+
+    def refreshed_form(self, form, work):
+        return self.Form(data={"commencing_work": form.cleaned_data["commencing_work"]}, instance=work)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["commencing_work"] = context["form"].data["commencing_work"]
+        return context
+
+
+class FindPublicationDocumentView(PlaceViewBase, TemplateView):
+    template_name = 'indigo_api/_work_possible_publication_documents.html'
+
+    class Form(forms.Form):
+        publication_date = forms.DateField()
+        publication_number = forms.CharField(required=False)
+        publication_name = forms.CharField()
```

Comment:
```suggestion
        publication_name = forms.CharField(required=False)
```
There's a default for publication name